src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -54,6 +49,22 @@ public abstract class BuildIntegratedNuGetProject : NuGetProject, INuGetIntegrat
             INuGetProjectContext projectContext,
             bool throwOnFailure);
 
+        public abstract Task<bool> InstallPackageAsync(
+            PackageIdentity packageIdentity,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<NuGetFramework> successfulFrameworks,
+            IEnumerable<NuGetFramework> unsucessfulFrameworks,
```

Comment:
Should there be another overload that doesn't take frameworks and applies it for all?